File: 01-backend/src/main/java/ee/insa/crmapp/model/InsurerTypeProfitPercent.java
package ee.insa.crmapp.model;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "insurer_type_profit_percent")
public class InsurerTypeProfitPercent {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;
    @OneToOne
    private Insurer insurer;
    @OneToOne
    private PolicyType policyType;
    @Column(name = "percent", nullable = false, precision=10, scale=2)
    private BigDecimal percent;
}

File: 02-frontend/src/app/model/insurertypeprofitpercent.ts
import {Insurer} from "./insurer";
import {PolicyType} from "./policytype";

export class InsurerTypeProfitPercent {
  public id: number;
  public insurer: Insurer;
  public policyType: PolicyType;
  public percent: number;

  constructor() {
    this.id = 0;
    this.insurer = null;
    this.policyType = null;
    this.percent = 0;
  }
}
